### PR TITLE
133 add alias to layer panel

### PR DESCRIPTION
**Pull request by coyle5280**

### CODE REVIEW

File: lib/carto/http_header_authentication.rb [hunk 1]
```diff
@@ -85,7 +85,7 @@ def field_from_value(request)
 
     def header_value(headers)
       header = ::Cartodb.get_config(:http_header_authentication, 'header')
-      puts "user-auto-creation : Trying to extract value from headers for #{header}, value is #{headers[header]}"
+      puts "user-auto-creation : Trying to extract value from headers for #{header}, value is #{headers[header || '']}"
```
Comment (gfiorav) [inline lib/carto/http_header_authentication.rb]: Don't commit this file! It's on a separate PR, not merged yet :)

File: lib/assets/javascripts/cartodb/table/layer_panel_view.js [hunk 2]
```diff
@@ -71,6 +70,10 @@
         this.$el.attr('model-id', this.model.id);
       }, this);
       this.model.bind('change:type',    this._setLayerType, this);
+
+      this.table.bind('update_layer_alias', this.addTableAlias, this);
+
+
```
Comment (gfiorav) [inline lib/assets/javascripts/cartodb/table/layer_panel_view.js]: I think it's a good idea to never leave two lines empty between statements!

File: lib/assets/javascripts/cartodb/table/layer_panel_view.js [hunk 3]
```diff
@@ -297,7 +306,7 @@
 
         // table name
         this.$('.layer-info .info .name')
-          .text(layer.get('table_name_alias') || layer.get('table_name').replace(/_/g,' '));
+          .text(this.table_alias_name || layer.get('table_name_alias') || layer.get('table_name').replace(/_/g,' '));
```
Comment (gfiorav) [inline lib/assets/javascripts/cartodb/table/layer_panel_view.js]: Interesting! When do these two hold different values?

```js
this.table_alias_name || layer.get('table_name_alias')
```

**Review by gfiorav — CHANGES_REQUESTED** [review]
Nice!

I have a couple of comments that need to be addressed.

Also, a question: will this happen when you're in dataset view? (i.e. 'Editing Mode')

Thanks!

File: lib/assets/javascripts/cartodb/table/layer_panel_view.js (hunk 3 above)
Comment (gfiorav) [reply lib/assets/javascripts/cartodb/table/layer_panel_view.js]: If `layer.get('table_name_alias')` points to nothing, we should remove it! 🔥

**Comment by gfiorav** [comment]
I'm getting an inconsistent behaviour in 'Editing Mode'


## STR:

0. Have a dataset with a table name alias
1. Go into 'Editing Mode' (`/user/<username>/tables/<table name>`)
2. Switch to 'Map'
3. Switch to 'Table'

## Expected result

The panel layer name is the aliased name

## Actual result

The panel layer name reverted to the non-aliased name

**Comment by gfiorav** [comment]
What's the status on this? Could you reproduce it or fix it?

**Comment by gfiorav** [comment]
Hey @coyle5280! Please take care of my previous comments and un commit the cdb module!

**Comment by coyle5280** [comment]
@gfiorav The cdb module has been rolled back to commit 52b4941 and the cdb file is longer in the PR.

**Review by gfiorav — APPROVED** [review]

**Pull request merged**